src/yz_index.erl

```diff
@@ -338,3 +374,45 @@ index_dir(Name) ->
 make_info(SchemaName, NVal) ->
     #index_info{n_val=NVal,
                 schema_name=SchemaName}.
+
+-spec sync_index(pid(), index_name(), timeout()) -> ok | timeout |
+                                                   {core_error, binary()}.
+sync_index(Pid, IndexName, Timeout) ->
+    process_flag(trap_exit, true),
+    {ok, Ring} = riak_core_ring_manager:get_my_ring(),
+    Nodes = riak_core_ring:all_members(Ring),
+    WaitPid = spawn_link(?MODULE, wait_for_index, [self(), IndexName, Nodes]),
+    register(wait_for_index, WaitPid),
+    receive
+        {_From, ok} ->
+            Pid ! ok;
+        {'EXIT', _Pid, _Reason} ->
+            sync_index(Pid, IndexName, Timeout)
+    after Timeout ->
+            exit(whereis(wait_for_index), kill),
+
+            %% Check if initFailure occurred
+            {ok, _, S} = yz_solr:core(status, [{wt,json},{core, IndexName}]),
+            case ?INITFAILURES(IndexName, S) of
+                [] ->
+                    lager:warning("Index ~s not created within ~p ms timeout",
```

Comment:
Can this be a notice instead of a warning? Timeouts are mildly interesting to the operator, but not necessarily indicative of an event which will become a problem if not corrected (i.e. could just be a little bit of overload, or maybe a slightly too short timeout).